api/server/main.go

```diff
@@ -0,0 +1,188 @@
+package main
+
+import (
+	"bytes"
+	"log"
+	"os"
+	"text/template"
+
+	"go/ast"
+	"go/parser"
+	"go/printer"
+	"go/token"
+
+	"github.com/pb33f/libopenapi"
+	v3high "github.com/pb33f/libopenapi/datamodel/high/v3"
+	"golang.org/x/tools/imports"
+)
+
+// exampleText defines the template in which the corresponding ExampleClien4_* body is wrapped.
+const exampleText = `
+package main
+
+import (
+{{- range .Imports -}}
+{{- if .}}
+{{"\t"}}{{.}}
+{{- else}}
+{{"\t"}}{{end -}}
+{{- end}}
+)
+
+func main() {
+{{.Body -}}
+}`
+
+func main() {
+	var exampleTmpl = template.Must(template.New("example").Parse(exampleText))
+
+	if len(os.Args) <= 1 {
+		log.Fatal("Expected filename to APIv4 spec as argument")
+	}
+
+	filename := os.Args[1]
+	data, err := os.ReadFile(filename)
+	if err != nil {
+		log.Fatalf("failed to read %s: %s", filename, err)
+	}
+
+	// Parse the Open APIv4 Spec
+	document, err := libopenapi.NewDocument(data)
+	if err != nil {
+		log.Fatalf("Failed to parse OpenAPI spec: %s", err)
+	}
+
+	v3Model, errors := document.BuildV3Model()
+	if len(errors) > 0 {
+		for i := range errors {
+			log.Printf("error: %s\n", errors[i])
+		}
+		log.Fatalf("cannot create v3 model from document: %d errors reported", len(errors))
+	}
+
+	applyExamples(v3Model, exampleTmpl)
+
+	// Re-render the file with the injected examples.
+	newDocument, _, _, errors := document.RenderAndReload()
+	if len(errors) > 0 {
+		for i := range errors {
+			log.Printf("error: %s\n", errors[i])
+		}
```

Comment:
Nit levels over 9000:
```suggestion
		for _, err := range errors {
			log.Printf("error: %s\n", err)
		}
```